### PR TITLE
TERRAM-6: standard .gitignore file

### PR DESCRIPTION
**Pull request by thedevilsvoice**
## Description

Security related change as described here: 
https://jira.paloaltonetworks.com:8443/secure/RapidBoard.jspa?rapidView=678&view=detail&selectedIssue=TERRAM-6

## Motivation and Context

Reduce chances that  developers will commit files and credentials to revision control system.

## How Has This Been Tested?

Reviewed by scrum team.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,29 @@
-.vscode
+# This is a minimal list, OK to add things per repo
+*.code-workspace
+*.plan
+*.tfstate*
```
Comment (damianfedeczko) [inline .gitignore:4]:
```suggestion
*.tfstate
*.tfstate*
```

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,29 @@
...
+.gitconfig
+.local
+# .netrc contains secrets for service tokens
+.netrc
+.sentinel
+# .ssh dir may contain private keys
+.ssh
+.terrascan
+# TF state stuff
+.terraform
```
Comment (damianfedeczko) [inline .gitignore:23]:
```suggestion
.terraform
.terraformrc
**/.terraform/*
terraform.rc
```

**Review by damianfedeczko — COMMENTED** [review]
Added some minor suggestions to think about, besides that - 🥇

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,29 @@
...
+.terraform.d
+.vscode
```
Comment (kbechler) [inline .gitignore:25]:
```suggestion
.vscode
.idea
```

Comment (kbechler) [inline .gitignore:25]:
```suggestion
.vscode
.idea
```

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,29 @@
...
+# Palo auth codes
+authcodes
+# Palo specific
+init-cfg.txt
```
Comment (kbechler) [inline .gitignore:29]:
```suggestion
init-cfg.txt
temp
tmp
```

**Review by kbechler — APPROVED** [review]
LGTM
+small suggestions

**Review by damianfedeczko — APPROVED** [review]

File: .gitignore [hunk 5]
```diff
@@ -1,2 +1,37 @@
-.vscode
+# This is a minimal list, OK to add things per repo
+*.code-workspace
+*.plan
+*.tfstate
+*.tfstate*
```
Comment (migara) [inline .gitignore:5]:
```suggestion
*.tfstate.*
```

File: .gitignore [hunk 6]
```diff
@@ -1,2 +1,37 @@
...
+.terraform.d
+.vscode
+.idea
+.idea
+# Palo auth codes
+authcodes
+# Palo specific
+init-cfg.txt
+temp
+tmp
```
Comment (migara) [inline .gitignore:37]:
```suggestion
tmp
credentials.json

# Crash log files
crash.log
```

**Review by migara — APPROVED** [review]
LGTM 👍🏽

**Review by damianfedeczko — APPROVED** [review]

**Pull request merged**